### PR TITLE
Add build workflow via docker

### PR DESCRIPTION
**Pull request by aktech**
Adds build workflow via docker based on the instructions specified here: https://github.com/tensorflow/data-validation#build-with-docker

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      uses: actions/checkout@v4
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Upgrade pip
+      run: |
+        python -m pip install --upgrade pip
```
Comment (peytondmurray) [inline .github/workflows/build.yml:29]: I guess we don't need python setup if we are just building this inside a docker container. Though maybe it makes sense if the wheel is being installed?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        docker compose run -e PYTHON_VERSION=$(echo "$version" | sed 's/\.//') manylinux2010
+
+    - name: Upload wheel artifact for Python ${{ matrix.python-version }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: data-validation-wheel-py${{ matrix.python-version }}
+        path: dist/*.whl
+
+    - name: Install built wheel
+      run: pip install dist/*.whl
```
Comment (peytondmurray) [inline .github/workflows/build.yml:46]: Is this just to check whether the wheel can be installed? If so maybe it makes sense to have a `twine check dist/*`?

**Review by peytondmurray — CHANGES_REQUESTED** [review]

**Comment by peytondmurray** [comment]
I think we also need to add the part of the workflow that automatically uploads to PyPI, like this https://github.com/tensorflow/tfx/pull/6892/files.

File: .github/workflows/build.yml (hunk 1 above)
Comment (aktech) [reply .github/workflows/build.yml:29]: >Though maybe it makes sense if the wheel is being installed?

Yep, that was the motivation for it.

**Review by peytondmurray — CHANGES_REQUESTED** [review]
Looks good, just need a `pip install twine` before running `twine check`.

**Review by smokestacklightnin — APPROVED** [review]

**Review by peytondmurray — APPROVED** [review]
Thanks!

**Pull request merged**